product_docs/docs/migration_portal/4/known_issues_notes.mdx

```diff
@@ -243,6 +245,59 @@ While using the Oracle default case, you may experience a lower compatibility ra
 -   The EDB DDL Extractor currently doesn't support the extraction of `ROLES`, `SYSTEM GRANTS ON ROLES`, `OBJECT GRANTS ON ROLES`, and `ROLE GRANTS` from Oracle 11g. This behavior results in error messages being written to the extracted files in the sections corresponding to these object types. These errors don't cause any issue in the assessment of these files by Migration Portal.
 -   The EDB DDL Extractor script may log `object "OBJECT_NAME" of type SYNONYM not found in schema "PUBLIC"` errors in the dependent objects section of the extracted file. This happens only if the user selects the option to extract dependent objects from an Oracle multi tenant environment where the Oracle database is a container database.
 
+### "Snapshot Too Old" Error
+
+The EDB DDL Extractor displays the error “ORA-01555: snapshot too old“ at runtime when the database server generates a volume of transactions that cannot be properly processed by the UNDO tablespace.
+
+When the database server generates undo transactions at a high rate, it can cause the server to run out of space to store undo data. Since the UNDO tablespace is implemented as a circular buffer, it starts overwriting older undo data blocks. This error allows the migration to continue, but may cause it to skip migrating public synonym database objects.
+
+To work around this error, increase the allocated space for the UNDO_RETENTION parameter.
+
+First, check the currently allocated total space for undo operations and the currently free space:
+
+1.  Check the existing value of the UNDO_RETENTION parameter:
+
+    ```sql
+    SELECT value FROM v$parameter WHERE name = 'undo_retention';
+    ```
+
+1.  Add the results of the two following queries to check the total available space in the UNDO tablespace:
+
+    ```sql
+    SELECT SUM(bytes)/1024/1024 free_mb FROM dba_undo_extents WHERE status='EXPIRED';
+    ```
+
+    ```sql
+    SELECT dfs.tablespace_name, SUM(dfs.bytes) / 1024 / 1024 AS free_mb FROM 
+    dba_free_space dfs 
+    JOIN dba_tablespaces dt ON dfs.tablespace_name = dt.tablespace_name 
+    WHERE dt.contents = 'UNDO' GROUP BY dfs.tablespace_name; 2 3 4
+    ```
+
+Next, determine the number of MB your environment required and increase the allocated storage space:
+
+1.  Monitor the volume of undo data generated by transactions:
+    
+    ```sql
+    SELECT SUM(undoblks * 8192)/1024/1024 AS undo_mbs FROM v$undostat;
```

Comment:
@gabbasb  
I fear the customer might be confused about what to do with this value. Here, we get a value of used MB (per minute? per hour? per 24 hours?), and in the next step, we tell them to multiply the MB/min value by the number of required retention minutes. 

According to my research, the default `v$undostat` view displays the used MB in the last 24 hours. Does that mean that we have to divide this value by 1440 (total minutes in 24 hours) to get the MB/min value? 

And then, in the next step, we would have to multiply that value by the number of minute we want undo data to be retained?

Reply:
@gvasquezvargas According to the docs
https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/V-UNDOSTAT.html
Each row in the view keeps statistics collected in the instance for a 10-minute interval.
The table holds start and end time as well.
However that's not the point.
What we are asking the user to do is to monitor the the total number of undo blocks consumed during peak usage hours using the V$UNDOSTAT.
This will give the user an idea of how much space the user needs to increase to avoid the error.